### PR TITLE
SOLR-17470: CLI resolve -s

### PR DESCRIPTION
**Pull request by epugh**
https://issues.apache.org/jira/browse/SOLR-17470

# Description

Please provide a short description of the changes you're making with this pull request.

# Solution

Please provide a short description of the approach taken to implement your solution.

# Tests

Please describe the tests you've developed or run to confirm this patch implements the feature or solves the problem.

# Checklist

Please review the following and check all that apply:

- [ ] I have reviewed the guidelines for [How to Contribute](https://github.com/apache/solr/blob/main/CONTRIBUTING.md) and my code conforms to the standards described there to the best of my ability.
- [ ] I have created a Jira issue and added the issue ID to my pull request title.
- [ ] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended, not available for branches on forks living under an organisation)
- [ ] I have developed this patch against the `main` branch.
- [ ] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

**Review by malliaridis — CHANGES_REQUESTED** [review]
Some hidden uses of `-s` that probably need to be updated as well:
- [x] `solr/dev-tools/scripts/cloud.sh#337` in `argsArray`: `-s` should be replaced with `--solr-home`
- [x] `solr/prometheus-exporter/bin/solr-exporter.cmd#81`: `-s` should be replaced with `--scrape-interval`
- [x] `solr/prometheus-exporter/bin/solr-exporter#115`: `-s` should be replaced with `--scrape-interval`
- `solr/solr-ref-guide/modules/deployment-guide/pages/solr-control-script-reference.adoc`
  - [x] `#214`: example should use `--solr-home`
  - [x] `#281`: -s should be `--solr-home` problably? 
- `solr/solr-ref-guide/modules/deployment-guide/pages/enabling-ssl.adoc`
  - [x] `#313`:  `-s` should be `--solr-home`
  - [x] `#322`: `-s` should be `--solr-home`
- `solr/solr-ref-guide/modules/getting-started/pages/tutorial-films.adoc`
  - [x] `#35`:  `-s` should be `--solr-home`
  - [x] `#43`: `-s` should be `--solr-home`
- `solr/solr-ref-guide/modules/getting-started/pages/tutorial-solrcloud.adoc`
  - [x] `#85`:  `-s` should be `--solr-home`
  - [x] `#179`: `-s` should be `--solr-home`
  - [x] `#186`: `-s` should be `--solr-home`
  - [x] `#199`: `-s` should be `--solr-home`
  - [x] `#209`: `-s` should be `--solr-home`
- `solr/solr-ref-guide/modules/getting-started/pages/tutorial-techproducts.adoc`
  - [x] `#74`:  `-s` should be `--solr-home`
  - [x] `#81`: `-s` should be `--solr-home`

Additional cases where we may consider replacing `-s` in the future:
- `solr/modules/hdfs/bin/prepare-snapshot-export.sh#46`: The option `-s` is used to provide a source path
       we may consider not replacing it here, because the entrypoint is a different script

**Comment by epugh** [comment]
I really like how the `--solr-home` parameter looks in the docs.  It's much less mysterious what is happening with the full parameter name used in the tutorials.  We may want to use the long option in our tutorials a lot more...

File: solr/solr-ref-guide/modules/deployment-guide/pages/solr-control-script-reference.adoc [hunk 1]
```diff
@@ -1205,7 +1205,7 @@ Is not required if `ZK_HOST` is defined in `solr.in.sh` or `solr.in.cmd`.
 +
 *Example*: `-z 123.321.23.43:2181`
 
-`-url <url>` or `--solr-url <url>`::
+`-s <url>` or --solr-url <url>`::
```
Comment (malliaridis) [inline solr/solr-ref-guide/modules/deployment-guide/pages/solr-control-script-reference.adoc:1208]:
```suggestion
`-s <url>` or `--solr-url <url>`::
```

File: solr/solr-ref-guide/modules/deployment-guide/pages/solr-control-script-reference.adoc [hunk 2]
```diff
@@ -1277,7 +1277,7 @@ Unnecessary if `ZK_HOST` is defined in `solr.in.sh` or `solr.in.cmd`.
...
```
Comment (malliaridis) [inline solr/solr-ref-guide/modules/deployment-guide/pages/solr-control-script-reference.adoc:1280]:
```suggestion
`-s <url>` or `--solr-url <url>`::
```

File: solr/solr-ref-guide/modules/deployment-guide/pages/solr-control-script-reference.adoc [hunk 3]
```diff
@@ -1361,7 +1361,7 @@ Optional if `ZK_HOST` is defined in `solr.in.sh` or `solr.in.cmd`.
...
```
Comment (malliaridis) [inline solr/solr-ref-guide/modules/deployment-guide/pages/solr-control-script-reference.adoc:1364]:
```suggestion
`-s <url>` or `--solr-url <url>`::
```

File: solr/solr-ref-guide/modules/deployment-guide/pages/solr-control-script-reference.adoc [hunk 4]
```diff
@@ -1444,7 +1444,7 @@ Optional if `ZK_HOST` is defined in `solr.in.sh` or `solr.in.cmd`.
...
```
Comment (malliaridis) [inline solr/solr-ref-guide/modules/deployment-guide/pages/solr-control-script-reference.adoc:1447]:
```suggestion
`-s <url>` or `--solr-url <url>`::
```

File: solr/solr-ref-guide/modules/deployment-guide/pages/solr-control-script-reference.adoc [hunk 5]
```diff
@@ -1506,7 +1506,7 @@ An example of this command is:
 
 `bin/solr zk mv /configs/oldconfigset /configs/newconfigset`
 
-`-url <url>` or `--solr-url <url>`::
+`-s <url>` or --solr-url <url>`::
```
Comment (malliaridis) [inline solr/solr-ref-guide/modules/deployment-guide/pages/solr-control-script-reference.adoc:1509]:
```suggestion
`-s <url>` or `--solr-url <url>`::
```

File: solr/solr-ref-guide/modules/deployment-guide/pages/solr-control-script-reference.adoc [hunk 6]
```diff
@@ -1557,7 +1557,7 @@ Optional if `ZK_HOST` is defined in `solr.in.sh` or `solr.in.cmd`.
 +
 *Example*: `-z 123.321.23.43:2181`
 
-`-url <url>` or `--solr-url <url>`::
+`-s <url>` or --solr-url <url>`::
```
Comment (malliaridis) [inline solr/solr-ref-guide/modules/deployment-guide/pages/solr-control-script-reference.adoc:1560]:
```suggestion
`-s <url>` or `--solr-url <url>`::
```

File: solr/solr-ref-guide/modules/deployment-guide/pages/solr-control-script-reference.adoc [hunk 7]
```diff
@@ -1606,7 +1606,7 @@ Optional if `ZK_HOST` is defined in `solr.in.sh` or `solr.in.cmd`.
...
```
Comment (malliaridis) [inline solr/solr-ref-guide/modules/deployment-guide/pages/solr-control-script-reference.adoc:1609]:
```suggestion
`-s <url>` or `--solr-url <url>`::
```

File: solr/solr-ref-guide/modules/deployment-guide/pages/solr-control-script-reference.adoc [hunk 8]
```diff
@@ -1786,7 +1786,7 @@ If you have configured basicAuth or TLS with your Solr you may find this easier
 
 The `bin/solr api` command takes the following parameters:
 
-`-url <url>` or `--solr-url <url>`::
+`-s <url>` or --solr-url <url>`::
```
Comment (malliaridis) [inline solr/solr-ref-guide/modules/deployment-guide/pages/solr-control-script-reference.adoc:1789]:
```suggestion
`-s <url>` or `--solr-url <url>`::
```

File: solr/solr-ref-guide/modules/deployment-guide/pages/solr-control-script-reference.adoc [hunk 9]
```diff
@@ -848,7 +848,7 @@ Skip safety checks when deleting the configuration directory used by a collectio
 The ZooKeeper connection string, usable in SolrCloud mode.
 Unnecessary if `ZK_HOST` is defined in `solr.in.sh` or `solr.in.cmd`.
 
-`-solr-url <url>`::
+`-s <url>` or --solr-url <url>`::
```
Comment (malliaridis) [inline solr/solr-ref-guide/modules/deployment-guide/pages/solr-control-script-reference.adoc:851]:
```suggestion
`-s <url>` or `--solr-url <url>`::
```

File: solr/solr-ref-guide/modules/deployment-guide/pages/solr-control-script-reference.adoc [hunk 10]
```diff
@@ -729,7 +729,7 @@ It is possible to override this warning with the --force parameter.
 The ZooKeeper connection string, usable in SolrCloud mode.
 Unnecessary if `ZK_HOST` is defined in `solr.in.sh` or `solr.in.cmd`.
 
-`--solr-url <url>`::
+`-s <url>` or --solr-url <url>`::
```
Comment (malliaridis) [inline solr/solr-ref-guide/modules/deployment-guide/pages/solr-control-script-reference.adoc:732]:
```suggestion
`-s <url>` or `--solr-url <url>`::
```

File: solr/solr-ref-guide/modules/deployment-guide/pages/solr-control-script-reference.adoc [hunk 11]
```diff
@@ -1094,7 +1094,7 @@ Set the property to this value; accepts JSON objects and strings.
 The ZooKeeper connection string, usable in SolrCloud mode.
 Unnecessary if `ZK_HOST` is defined in `solr.in.sh` or `solr.in.cmd`.
 
-`-u <url>` or --solr-url <url>`::
+`-s <url>` or --solr-url <url>`::
```
Comment (malliaridis) [inline solr/solr-ref-guide/modules/deployment-guide/pages/solr-control-script-reference.adoc:1097]:
```suggestion
`-s <url>` or `--solr-url <url>`::
```

File: solr/solr-ref-guide/modules/deployment-guide/pages/solr-control-script-reference.adoc [hunk 12]
```diff
@@ -548,7 +548,7 @@ Name of the collection to run a healthcheck against.
 +
 *Example*: `bin/solr healthcheck -c gettingstarted`
 
-`--solr-url <url>`::
+`-s <url>` or --solr-url <url>`::
```
Comment (malliaridis) [inline solr/solr-ref-guide/modules/deployment-guide/pages/solr-control-script-reference.adoc:551]:
```suggestion
`-s <url>` or `--solr-url <url>`::
```

**Review by malliaridis — CHANGES_REQUESTED** [review]
On second review, I noticed multiple occurences in bin/solr.cmd (I searched for "-s"). I am not sure how `-s` behaves when used, because the script uses `-s` for solr-home and sorl-url in the same blocks. I will try to test it on a windows machine before approval.

Could you have a look as well @epugh?

I also found for \`-s\` occurences in:
- [ ] solr/solr-ref-guide/modules/getting-started/pages/tutorial-films.adoc#85 -s should be --shards
- [ ] solr/solr-ref-guide/modules/deployment-guide/pages/solr-control-script-reference.adoc#108: -s should be --solr-home
- [ ] solr/solr-ref-guide/modules/deployment-guide/pages/monitoring-with-prometheus-and-grafana.adoc#162: `-s` option should not be included for `--scrape-interval`

**Review by malliaridis — APPROVED** [review]
Looks good and complete to me. :)

**Pull request merged**